mongo/mongo.go

```diff
@@ -106,13 +106,22 @@ func (m *Mongo) GetBooks(ctx context.Context) (models.Books, error) {
 
 	list := session.DB(m.Database).C(m.BooksCollection).Find(nil)
 
-	books := &models.Books{}
-	if err := list.All(&books.Items); err != nil {
+	totalCount, err := list.Count()
+	if err != nil {
+		log.Event(ctx, "error counting items", log.ERROR, log.Error(err))
+		if err == mgo.ErrNotFound {
+			return []*models.Book{}, totalCount, nil
+		}
+	}
```

Comment:
Missing return if there is an error returned from mongo that is not mgo.ErrNotFound

Also the log event could serve you the developer better by doing something like:

```
...
	totalCount, err := list.Count()
	if err != nil {
		if err == mgo.ErrNotFound {
			log.Event(ctx, "no book resources found", log.WARN, log.Error(err))
			return []*models.Book{}, totalCount, nil
		}

		log.Event(ctx, "failure to retrieve list of books", log.ERROR, log.Error(err))
		return nil, 0, err
	}
...
```